src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx

```diff
@@ -27,6 +29,37 @@ export const Form: React.FC<FormProps> = props => {
   const [isLoading, setIsLoading] = useState(false)
   const { state, dispatch } = props
   const { sendToast } = useToasts()
+  const { user } = useSystemContext()
+
+  useEffect(() => {
+    if (!user) {
+      return
+    }
+
+    const getWeaviateUser = () => {
+      return fetch(`/api/advisor/7/users/${user.id}`)
+    }
+
+    const createWeaviateUser = () => {
+      const headers = { "Content-Type": "application/json" }
+      const body = JSON.stringify({ userId: user.id, name: user.name })
+      const options = { method: "POST", headers, body }
+      return fetch(`/api/advisor/7/users`, options)
+    }
+
+    getWeaviateUser().then(response => {
+      if (response.status === 404) {
+        console.log("Creating user in Weaviate", response)
+        createWeaviateUser()
+      }
+
+      if (!response.ok) {
+        throw new Error(
+          `Error checking for weaviate user: ${response.statusText}`
+        )
+      }
+    })
+  }, [user])
```

Comment:
Some slightly awkward promise chaining here. But seemed the simplest approach. Happy to change if there is a preferred alternative.